circle.yml

```diff
@@ -6,37 +6,54 @@ machine:
 
 dependencies:
   pre:
+    # Decode encrypted secrets and release keys
     - openssl aes-256-cbc -d -in ci-keys.cipher -k $CI_SECRETS_KEY >> ~/.circlerc
     - openssl aes-256-cbc -d -in release.keystore.cipher -out release.keystore -k $CI_SECRETS_KEY
+    # Ensure we have the correct Android build environment
     - if [ ! -e $ANDROID_HOME/build-tools/$ANDROID_BUILD_TOOLS ]; then echo y | android update sdk --no-ui --all --filter build-tools-$ANDROID_BUILD_TOOLS; fi
     - if [ ! -e $ANDROID_HOME/platforms/android-$ANDROID_PLATFORM ]; then echo y | android update sdk --no-ui --all --filter "android-${ANDROID_PLATFORM}"; fi
+    # Ensure we have the latest Android support libraries
     - echo y | android update sdk -u -a -t extra-android-m2repository
     - echo y | android update sdk -u -a -t extra-google-m2repository
 
   cache_directories:
+    # Cache the build environment to speed up further builds
     - /usr/local/android-sdk-linux/build-tools/24.0.1
```

Comment:
We're caching the build environment which has hardcoded versions. Can we reuse the `$ANDROID_HOME/build-tools/$ANDROID_BUILD_TOOLS` and `$ANDROID_HOME/platforms/android-$ANDROID_PLATFORM` to make them more dynamic?